### PR TITLE
Add message indicator color option for website dark mode and customizable colors

### PR DESCRIPTION
**Pull request by actuallyronan**
### Changes

Adds a color setting to the "Website Dark Mode and Customizable Colors" add-on that lets you select the color for the message indicator.

### Reason for changes

Many people (including myself) would prefer to be able to change the color of the message indicator, as it is a key part of the Scratch website.

### Tests

Tested on Chrome 108 and it seems to work.

### CODE REVIEW

File: addons/dark-www/experimental_scratchr2.css [hunk 1]
```diff
@@ -671,3 +671,7 @@ input.link.black {
 .iframeshim {
   color-scheme: light;
 }
+
+.message-count, .notificationsCount {
+  background-color: var(--darkWww-message-indicator-color) !important; 
+}
```
Comment (mxmou) [inline addons/dark-www/experimental_scratchr2.css:677]:
```suggestion
#topnav ul.account-nav .messages .notificationsCount {
  background-color: var(--darkWww-message-indicator-color); 
}
```

File: addons/dark-www/experimental_scratchwww.css [hunk 2]
```diff
@@ -862,3 +862,7 @@ p {
   /* Change the solid color background to a transparent one */
   background-color: rgba(0, 165, 0, 0.07);
 }
+
+.message-count, .notificationsCount {
+  background-color: var(--darkWww-message-indicator-color) !important; 
+}
```
Comment (mxmou) [inline addons/dark-www/experimental_scratchwww.css:868]:
```suggestion
#navigation .messages .message-count.show {
  background-color: var(--darkWww-message-indicator-color); 
}
```

**Review by mxmou — CHANGES_REQUESTED** [review]
You don't need to define a `customCssVariable` for this - one will be created automatically based on the setting name (`--darkWww-messageIndicatorColor` in this case).

**Comment by Samq64** [comment]
Could you please enable actions? (https://github.com/actuallyronan/ScratchAddons/actions)

**Comment by DNin01** [comment]
> Could you please prettier on the actions page? (https://github.com/actuallyronan/ScratchAddons/actions)

I think it's not that Prettier isn't working, it's that it's not turning tab characters into two spaces.

@actuallyronan could you change all the tab characters to two spaces?

**Comment by actuallyronan** [comment]
> > Could you please prettier on the actions page? (https://github.com/actuallyronan/ScratchAddons/actions)
> 
> I think it's not that Prettier isn't working, it's that it's not turning tab characters into two spaces. The CGs already recommend enabling workflows, so they probably already enabled it.
> 
> @actuallyronan could you change all the tab characters to two spaces?

Done

File: addons/dark-www/addon.json [hunk 3]
```diff
@@ -1,2107 +1,2183 @@
...
+      "allowTransparency":true,
+      "default":"#606060"
     },
     {
-      "name": "Border color",
-      "id": "border",
-      "type": "color",
-      "allowTransparency": true,
-      "default": "#606060"
+      "name":"Message indicator color",
```
Comment (DNin01) [inline addons/dark-www/addon.json:2081]: The messages "indicator" is also known as the messages badge.

**Review by DNin01 — COMMENTED** [review]
> Done

Thanks!

Also, sorry to bug you _again_, but another formatting thing in the `addon.json` is that there is usually a space after the `:` right after each key (example: `"key": "value"`), so maybe change that as well, to keep the formatting consistent with other files.

**Comment by mxmou** [comment]
@actuallyronan Formatting is now fixed.

**Review by DNin01 — CHANGES_REQUESTED** [review]
Four things:
1. You ~~need to~~ might want to add the default message badge color to the other presets (look under `"presets": [`).
2. This doesn't apply to the messages page itself.[^1]
3. If you set the badge color to a bright color, the number on it should turn black.
4. This is generally known as the messages badge.

[^1]: See image: 
An orange messages badge on the Messages page

**Comment by mxmou** [comment]
> You need to add the default message badge color to the other presets (look under `"presets": [`).

This isn't actually necessary, but it's probably a good idea - I think it makes sense to reset the color when selecting a preset.

**Comment by Samq64** [comment]
Are there any other orange/yellow elements on the site that could be customized with this setting?

**Comment by DNin01** [comment]
> Are there any other orange/yellow elements on the site that could be customized with this setting?

The Share button.

**Comment by DNin01** [comment]
> > You need to add the default message badge color to the other presets (look under `"presets": [`).
> 
> This isn't actually necessary, but it's probably a good idea - I think it makes sense to reset the color when selecting a preset.

The reset menu for that color will look like this, though.

The preset color choices menu with transparent backgrounds from the presets

Clicking on one of them, however, still resets the color to the default, not a transparent background as shown.

**Comment by mxmou** [comment]
> > > You need to add the default message badge color to the other presets (look under `"presets": [`).
> > 
> > 
> > This isn't actually necessary, but it's probably a good idea - I think it makes sense to reset the color when selecting a preset.
> 
> The reset menu for that color will look like this, though.
> 
> The preset color choices menu with transparent backgrounds from the presets
> 
> Clicking on one of them, however, still resets the color to the default, not a transparent background as shown.

That's a bug with the settings page.

**Comment by WorldLanguages** [comment]
> That's a bug with the settings page.

Could you create a GitHub issue about it? (I think it doesn't exist yet)

**Comment by mxmou** [comment]
@WorldLanguages I'll just open a PR to fix it - it's a very simple change.

**Comment by mxmou** [comment]
If a light color is chosen, the message count should become black. You can use `customCssVariables` to implement this.

**Comment by WorldLanguages** [comment]
@mxmou We should probably move this to the v1.32 milestone?

**Comment by Secret-chest** [comment]
What about the Explore orange?

**Comment by mxmou** [comment]
> What about the Explore orange?

That should probably be in a separate addon for customizing banner colors, similar to #4187.

**Comment by WorldLanguages** [comment]
What's the status of this PR?

**Comment by mxmou** [comment]
> What's the status of this PR?

This feature is still missing:

> If a light color is chosen, the message count should become black. You can use `customCssVariables` to implement this.

**Comment by DNin01** [comment]
Scratch changed the message counter background in the update. Do we need to worry about making it customizable anymore? If not, I'll close this.

**Comment by DNin01** [comment]
Well, there is the fact that some people might not like it to be dark red.

This PR isn't being actively maintained right now, so maybe we can start a new one if anyone is interested.

**Comment by WorldLanguages** [comment]
> Well, there is the fact that some people might not like it to be dark red.

There were one of two people sending feedback about this. Maybe it could be part of this addon.

**Comment by mxmou** [comment]
> > Well, there is the fact that some people might not like it to be dark red.
> 
> There were one of two people sending feedback about this. Maybe it could be part of this addon.

The option this PR currently adds changes both counters - the one in the navigation bar and the one on the Messages page. It should probably be split into two.

**Comment by DNin01** [comment]
> > > Well, there is the fact that some people might not like it to be dark red.
> 
> > 
> 
> > There were one of two people sending feedback about this. Maybe it could be part of this addon.
> 
> 
> 
> The option this PR currently adds changes both counters - the one in the navigation bar and the one on the Messages page. It should probably be split into two.

That would work, but it feels like too much. I wonder how we could implement this while only adding one color to the settings.

**Comment by WorldLanguages** [comment]
> I don't think there's a (good) way to make both colors customizable with just one setting.

I agree. It would be nice if we could provide a `Scratch default colors (blue)` preset (or maybe another name) for this addon.
@mxmou It's up to you if you want to work on this PR or create a new one

**Comment by DNin01** [comment]
I don't feel like customizing the badge above the list of messages counting the number of unread messages is as important. If we were to go with a one-setting method, I'd allow customizing the badge on the navigation bar and not the one on the messages page.

**Comment by WorldLanguages** [comment]
From feedback:

> you forgot the option to change the message notifier from a putrid orange red to any color

**Comment by mxmou** [comment]
> I don't feel like customizing the badge above the list of messages counting the number of unread messages is as important. If we were to go with a one-setting method, I'd allow customizing the badge on the navigation bar and not the one on the messages page.

People want to change the one on the messages page back to orange though.

**Review by mxmou — APPROVED** [review]

**Comment by WorldLanguages** [comment]
Looks good!
Does this take into account that the scratchr2 message button orange is a slightly different shade? (Not saying it should, but I've checked whether they have the same exact hex code). Looks like it does. Nice!

**Comment by WorldLanguages** [comment]
I guess this will need a settings migration for users who already selected these presets (specifically, the ones with `(blue)` in the name). I'd say we create an issue after merging this and deal with it later.

**Comment by WorldLanguages** [comment]
(We should also add a latestUpdate so that users notice this update)

**Comment by DNin01** [comment]
Alright, this seems like the best we can do, and that's fine.

**Review by WorldLanguages — APPROVED** [review]

**Pull request merged**